### ISSUE TITLE
Gradients from automatic differentiation 

### ISSUE THREAD

**Issue by LeonardoBarneschi** [learningmatter-mit/NeuralForceField#19]
Dear developers, 
I apologize in advance for my question, which may be trivial. I would like to train a differentiable NN model on energies, with gradient from automatic differentiation. I don't want to train directly on gradients, but I would like to get them as energy derivative. Basically I will have my own dataset to train the model on energies, get gradients from the ML model differentiation, and then compare them with a few gradients calculated from very expensive QM calculations. That's indeed why I cannot train directly on gradients, it would be impossible for me to create a sizeable training set.
In case this is possible, could you provide a minimal example or reference? I just started using nff. 
 
Thank you in advance.

Best regards,
Leo

**Comment by simonaxelrod** [comment]
Hi Leo,

Thanks for your question! To train on just energies and not gradients, you can (for example) make a PaiNN model by following [tutorial 9](https://github.com/learningmatter-mit/NeuralForceField/blob/master/tutorials/09_painn.ipynb), but replace this line
```python
loss_fn = loss.build_mse_loss(loss_coef={'energy_grad': 0.95, 'energy': 0.05})
```
with this
```python
loss_fn = loss.build_mse_loss(loss_coef={'energy': 0.05})
```
Now the model will only be trained on energies. However, the model will still take gradients for the keys in `modelparams['grad_keys']`. In the notebook we set `grad_keys` to `["energy_grad"]`, so the model will compute the gradient of the energy after each forward pass, but it will not use it in the loss function. So training with this loss function will work for you, but it will be unnecessarily slow. To save time during training, you'd therefore want to also set `grad_keys` to `[]` in `modelparams`. 

`grad_keys` gets stored in the model as an attribute, `model.grad_keys`. To compute gradients for a small subset of molecules after training, you can set `model.grad_keys = ["energy_grad"]`. Then when you call the model on your batch of molecules, it will automatically compute gradients as well.

Hope this helps!

Simon
